src/sim/UI/ViewClass.py

```diff
@@ -186,6 +198,45 @@ def __init__(self, target, args):
 
         super().__init__()
 
+
     @pyqtSlot()
     def run(self):
-        self.__target(self.__args)
\ No newline at end of file
+        self.__target(self.__args)
+
+
+# Custom QGraphicsEllipseItem
+class NamedEllipseItem(QGraphicsEllipseItem):
+    def __init__(self, name, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.__name = name
+
+
+    def getName(self):
+        return self.__name
+    
+
+# TODO: I don't like having to pass the controller to this class, need to update to find a work around for that
+class CustomQGraphicsScene(QGraphicsScene):
+    def __init__(self, controller, parent=None):
+        super().__init__(parent)
+        self.__controller = controller
+
+    def mouseReleaseEvent(self, event):
+        # Call the base class method to handle normal mouse press events
+        super().mouseReleaseEvent(event)
+        
+        # Custom release handler TODO: UPDATE TO MOVE PROCESSING TO WORKER
+        self.__CustomReleaseHandler(event)
+
+
+    # Custom release handler to call controller and update robot positions
+    def __CustomReleaseHandler(self, event):
+        if event.button() == Qt.LeftButton:
+            mousePos = event.scenePos()
+            items = self.items(mousePos)
+            for item in items:
+                # Need check because items contains both Ellipse & Text but Ellipse is the only one with updated coordinates
+                if (item.__class__.__name__ == "NamedEllipseItem"):
+                    # update the robots x,y coordinate in robot_positions
+                    self.__controller.updateRobotPositions(item.getName(), item.x(), item.y())
```

Comment:
these fields (x and y) do not need to be used, they can be deleted and the event handler removed. The [QPointF](https://doc.qt.io/qt-6/qpointf.html) has the location of the ellipse all the time. sorry, thought I told you about this before, I must have forgotten

Reply:
We discussed this briefly in passing so I don't think all the information came across. I'll investigate this to incorporate QPointF instead of the item properties and event handler

Reply:
Updated as of commit (8ef8ad5), review and resolve if approved